### PR TITLE
Add Flake8 plugin for class attributes

### PR DESCRIPTION
**Pull request by PelegLi**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Added a flake8 plugin to warn from creating class attributes instead of instance attributes.
Currently all folders are excluded.

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -4,3 +4,10 @@
 max-line-length = 120
 # E501 doesn't work with black together
 ignore = E127,E128,E129,E203,E231,E302,E303,E305,E501,E731,W291,W292,W293,W503,W504,DUO107,DUO104,DUO130,DUO109,DUO116
+extend-exclude = .github, .pytest_cache, admissioncontroller/*, bin/*, docs/*, github_action_resources/*, integration_tests/*, kubernetes/*, performance_tests/*, venv/*, tests/*, flake8_plugins/*, checkov/arm/*, checkov/bitbucket/*, checkov/bitbucket_pipelines/*, checkov/cloudformation/*, checkov/common/*, checkov/dockerfile/*, checkov/github/*, checkov/github_actions/*, checkov/gitlab/*, checkov/gitlab_ci/*, checkov/json_doc/*, checkov/kubernetes/*, checkov/sca_image/*, checkov/sca_package/*, checkov/secrets/*, checkov/serverless/*, checkov/terraform/*, checkov/yaml_doc/*, checkov/openapi/*, checkov/example_runner/*, checkov/bicep/*, checkov/kustomize/*, checkov/helm/*
```
Comment (achiar99) [inline .flake8:7]: dont we want to fix all the static attributes?

File: flake8_plugins/flake8_class_attributes_plugin/flake8_class_attributes/class_members_errors.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+def get_node_name(node, node_type: str):
+    special_methods_names = (
+        '__new__',
+        '__init__',
+        '__post_init__',
+        '__str__',
+        'save',
+        'delete',
+    )
+    name_getters_by_type = [
```
Comment (achiar99) [inline flake8_plugins/flake8_class_attributes_plugin/flake8_class_attributes/class_members_errors.py:28]: we just care about the 'field' here no? why do we need all the other stuff?

File: flake8_plugins/flake8_class_attributes_plugin/tests/test_class_attributes.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+from conftest import run_validator_for_test_file
+
+
+def test_file_with_class_attribute():
+    errors = run_validator_for_test_file('class_attribute_fail.py')
+    assert len(errors) == 1
```
Comment (achiar99) [inline flake8_plugins/flake8_class_attributes_plugin/tests/test_class_attributes.py:6]: 👏

**Review by achiar99 — APPROVED** [review]
very cool!

File: .flake8 (hunk 1 above)
Comment (gruebel) [reply .flake8:7]: how about just including one of the checkov sub folders, or is one already included for checking

Comment (PelegLi) [reply .flake8:7]: @rotemavni is currently refactoring the whole project according to this check. Once she commits the fixes we can remove most of the excludes

**Review by rotemavni — APPROVED** [review]

**Pull request merged**